src/metabase/driver/h2.clj

```diff
@@ -573,13 +573,14 @@
 (defmethod driver/upload-type->database-type :h2
   [_driver upload-type]
   (case upload-type
-    ::upload/varchar_255 "VARCHAR"
-    ::upload/text        "VARCHAR"
-    ::upload/int         "BIGINT"
-    ::upload/float       "DOUBLE PRECISION"
-    ::upload/boolean     "BOOLEAN"
-    ::upload/date        "DATE"
-    ::upload/datetime    "TIMESTAMP"))
+    ::upload/varchar_255 :varchar
+    ::upload/text        :varchar
+    ::upload/int         :bigint
+    ::upload/pk          :int
```

Comment:
PKs aren't restricted to int types. e.g. UUIDs are strings. Have you considered how to extend this to support PKs with string types? Would you change this to `::upload/pk-int` and introduce `::upload/pk-varchar_255`?

Reply:
yes, but not for milestone 1